compiler/luci/pass/src/FuseGRUPass.cpp

```diff
@@ -0,0 +1,674 @@
+{
+  auto cloned = graph->nodes()->create<luci::CircleConst>();
+
+  if (cloned != nullptr)
+  {
+    // dtype/shape
+    cloned->dtype(node->dtype());
+    cloned->rank(node->rank());
+
+    // values
+    switch (node->dtype())
+    {
+      case loco::DataType::FLOAT32:
+        copy_values<loco::DataType::FLOAT32>(node, cloned);
+        break;
+
+      case loco::DataType::U8:
+        copy_values<loco::DataType::U8>(node, cloned);
+        break;
+
+      case loco::DataType::S8:
+        copy_values<loco::DataType::S8>(node, cloned);
+        break;
+
+      case loco::DataType::S16:
+        copy_values<loco::DataType::S16>(node, cloned);
+        break;
+
+      case loco::DataType::S32:
+        copy_values<loco::DataType::S32>(node, cloned);
+        break;
+
+      case loco::DataType::S64:
+        copy_values<loco::DataType::S64>(node, cloned);
+        break;
+
+      case loco::DataType::BOOL:
+        copy_values<loco::DataType::BOOL>(node, cloned);
+        break;
+
+      default:
+        throw std::runtime_error("FuseGRU: Unsupported data type");
+    }
+  }
+
+  return cloned;
+}
+
+luci::CircleGRU *FuseGRU::create_circle_gru(loco::Graph *graph)
+{
+  assert(graph);
+
+  auto weight_ih_cloned = clone_circleconst(_p->_weight_ih, graph);
+  luci::copy_common_attributes(_p->_weight_ih, weight_ih_cloned);
+
+  auto weight_hh_cloned = clone_circleconst(_p->_weight_hh, graph);
+  luci::copy_common_attributes(_p->_weight_hh, weight_hh_cloned);
+
+  luci::CircleNode *bias_ih_cloned = nullptr;
+  if (_p->_bias_ih != nullptr)
+  {
+    bias_ih_cloned = clone_circleconst(_p->_bias_ih, graph);
+    luci::copy_common_attributes(_p->_bias_ih, bias_ih_cloned);
+  }
+  else
+  {
+    bias_ih_cloned = _p->_pattern_last_node->graph()->nodes()->create<luci::CircleOutputExclude>();
+  }
+
+  luci::CircleNode *bias_hh_cloned = nullptr;
+  if (_p->_bias_hh != nullptr)
+  {
+    bias_hh_cloned = clone_circleconst(_p->_bias_hh, graph);
+    luci::copy_common_attributes(_p->_bias_hh, bias_hh_cloned);
+  }
+  else
+  {
+    bias_hh_cloned = _p->_pattern_last_node->graph()->nodes()->create<luci::CircleOutputExclude>();
+  }
+
+  auto hidden_input_cloned = clone_circleconst(_p->_hidden_input, graph);
+  luci::copy_common_attributes(_p->_hidden_input, hidden_input_cloned);
+
+  auto less_const_cloned = clone_circleconst(_p->_less_const, graph);
+  luci::copy_common_attributes(_p->_less_const, less_const_cloned);
+
+  // Create and configure new CircleGRU operation.
+  auto circle_gru = _p->_while_node->graph()->nodes()->create<luci::CircleGRU>();
+  circle_gru->input(_p->_ifm);
+  circle_gru->hidden_hidden(weight_hh_cloned);
+  circle_gru->hidden_input(weight_ih_cloned);
+  circle_gru->hidden_hidden_bias(bias_hh_cloned);
+  circle_gru->hidden_input_bias(bias_ih_cloned);
+  circle_gru->state(hidden_input_cloned);
+
+  // Note: Now support only returnSequences = false
+  circle_gru->returnSequences(false);
+  circle_gru->name("FusedCircleGRU");
```

Comment:
If there are multiple GRU Ops in the model, nodes will have duplicate name, which is prevented in circle.

```suggestion
  circle_gru->name(_while_node->name() + "_FusedCircleGRU");
```